### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by LPeter1997**
BCL docs, array base types, enum equality

Known issues:
 * Array uses the nongeneric enumerators
 * Array indexing could be removed
 * Missing tests

### CODE REVIEW

File: src/Draco.Compiler/Internal/Documentation/Extractors/XmlDocumentationExtractor.cs [hunk 1]
```diff
@@ -93,13 +93,24 @@ private ImmutableArray<DocumentationElement> ExtractElementsFromNode(XmlNode nod
...
-        var symbol = this.GetSymbolFromDocumentationName(cref ?? string.Empty)
-            // NOTE: The first two characters of the link is the documentation prefix
-            ?? new PrimitiveTypeSymbol(cref?[2..] ?? string.Empty, false);
-        return new ReferenceDocumentationElement(symbol, string.IsNullOrEmpty(node.InnerText) ? null : node.InnerText);
+        if (cref is not null)
+        {
+            var symbol = this.GetSymbolFromDocumentationName(cref ?? string.Empty)
+                // NOTE: The first two characters of the link is the documentation prefix
+                ?? new PrimitiveTypeSymbol(cref?[2..] ?? string.Empty, false);
+            return new ReferenceDocumentationElement(symbol, string.IsNullOrEmpty(node.InnerText) ? null : node.InnerText);
```
Comment (Kuinox) [inline src/Draco.Compiler/Internal/Documentation/Extractors/XmlDocumentationExtractor.cs:105]: Simplify by just checking empty string.

File: src/Draco.Compiler/Internal/Symbols/Metadata/MetadataMethodSymbol.cs [hunk 2]
```diff
@@ -62,7 +62,14 @@ public override bool IsVirtual
                 || this.Override is not null;
         }
     }
+
     public override bool IsStatic => methodDefinition.Attributes.HasFlag(MethodAttributes.Static);
+
+    // TODO: Very hacky way of doing this
+    public override bool IsExplicitImplementation =>
+           this.Visibility == Api.Semantics.Visibility.Private
+        || this.Name.Contains('.');
```
Comment (Kuinox) [inline src/Draco.Compiler/Internal/Symbols/Metadata/MetadataMethodSymbol.cs:71]: What is an explicit implementation

File: src/Draco.Compiler/Internal/Documentation/Extractors/XmlDocumentationExtractor.cs [hunk 3]
```diff
@@ -93,13 +93,24 @@ private ImmutableArray<DocumentationElement> ExtractElementsFromNode(XmlNode nod
...
     {
+        // It can be a 'cref' or a 'langword'
         var cref = node.Attributes?["cref"]?.Value;
-        var symbol = this.GetSymbolFromDocumentationName(cref ?? string.Empty)
-            // NOTE: The first two characters of the link is the documentation prefix
-            ?? new PrimitiveTypeSymbol(cref?[2..] ?? string.Empty, false);
-        return new ReferenceDocumentationElement(symbol, string.IsNullOrEmpty(node.InnerText) ? null : node.InnerText);
+        if (cref is not null)
+        {
+            var symbol = this.GetSymbolFromDocumentationName(cref ?? string.Empty)
```
[the remark this answers is not in the document]
Comment (LPeter1997) [reply src/Draco.Compiler/Internal/Documentation/Extractors/XmlDocumentationExtractor.cs:102]: Pains of copypasta

File: src/Draco.Compiler/Internal/Symbols/Metadata/MetadataMethodSymbol.cs (hunk 2 above)
Comment (LPeter1997) [reply src/Draco.Compiler/Internal/Symbols/Metadata/MetadataMethodSymbol.cs:71]: Basically an interface member that gets hidden at the implementation site:
```cs
public class Foo : IList<int> {
    // other members...

    public IEnumerator<int> GetEnumerator() => /* ... */;

    // This is an explicit implementation
    IEnumerator IEnumerable.GetEnumerator() => /* ... */;
}
```

Comment (Kuinox) [reply src/Draco.Compiler/Internal/Symbols/Metadata/MetadataMethodSymbol.cs:71]: Ah yes, I know what are thoses, I just didn't remembered about this thing when I've read the name.

**Review by Kuinox — APPROVED** [review]

**Pull request merged**